core/src/main/scala/kafka/admin/ZkSecurityMigrator.scala

```diff
@@ -0,0 +1,209 @@
+          case _ =>
+            error("Unexpected return code: %d".format(rc))
+            exception = throw ZkException.create(KeeperException.create(rc))   
+      }
+    }
+  }
+  
+  def run(args: Array[String]) {
+    var jaasFile = System.getProperty(JaasUtils.JAVA_LOGIN_CONFIG_PARAM)
+    val parser = new OptionParser()
+
+    val jaasFileOpt = parser.accepts("jaas.file", "JAAS Config file.").withOptionalArg().ofType(classOf[String])
+    val zkUrlOpt = parser.accepts("connect", "Sets the ZooKeeper connect string (ensemble). This parameter " +
+                                  "takes a comma-separated list of host:port pairs.").
+      withRequiredArg().defaultsTo("localhost:2181").ofType(classOf[String])
+    val zkSessionTimeoutOpt = parser.accepts("session.timeout", "Sets the ZooKeeper session timeout.").
+      withRequiredArg().ofType(classOf[java.lang.Integer])
+    val zkConnectionTimeoutOpt = parser.accepts("connection.timeout", "Sets the ZooKeeper connection timeout.").
+      withRequiredArg().defaultsTo("30000").ofType(classOf[java.lang.Integer])
+    val helpOpt = parser.accepts("help", "Print usage information.")
+
+    val options = parser.parse(args : _*)
+    if(options.has(helpOpt))
+      CommandLineUtils.printUsageAndDie(parser, "ZooKeeper Migration Tool Help. This tool updates the ACLs of "
+                                                + "znodes as part of the process of setting up ZooKeeper "
+                                                + "authentication.")
+
+    if ((jaasFile == null) && !options.has(jaasFileOpt)) {
+      error("No JAAS configuration file has been specified. Please make sure that you have set either "
+            + "the system property %s or the option %s".format(JaasUtils.JAVA_LOGIN_CONFIG_PARAM, "--jaas.file"))
+      throw new IllegalArgumentException("Incorrect configuration")
+    }
+    
+    if (jaasFile == null) {
+      jaasFile = options.valueOf(jaasFileOpt)
+      System.setProperty(JaasUtils.JAVA_LOGIN_CONFIG_PARAM, jaasFile)
+    }
+    
+    if (!JaasUtils.isZkSecurityEnabled(jaasFile)) {
+      error("Security isn't enabled, most likely the file isn't set properly: %s".format(jaasFile))
+      throw new IllegalArgumentException("Incorrect configuration") 
+    }
+
+    val zkUrl = options.valueOf(zkUrlOpt)
+
+    /* TODO: Need to fix this before it is checked in, not compiling
+     * val zkSessionTimeout = if (options.has(zkSessionTimeoutOpt)) {
+       options.valueOf(zkSessionTimeoutOpt).intValue
+     } else {
+       30000
+     }
+     val zkConnectionTimeout = if (options.has(zkConnectionTimeoutOpt)) {
+       options.valueOf(zkConnectionTimeoutOpt).intValue
+     } else {
+       30000
+     }*/
+    val zkUtils = ZkUtils.apply(zkUrl, 30000, 30000, true)
+
+    for (path <- zkUtils.persistentZkPaths) {
+      zkUtils.makeSurePersistentPathExists(path)
+      setAclsRecursively(zkUtils, path)
```

Comment:
Yes, we need to exclude it. I'm thinking that we don't need to make sure that the consumers path is created in this tool, so I've created a second list for secure paths. I could have done it by keeping the `persistentZkPaths` and adding one for exclusion, but if I take this approach, then I need to keep checking it all the time so I've opted for a second list.

Also, I'm thinking of moving the two lists to `object ZkUtils` so that we don't have to instantiate them every time with `ZkUtils`, but that's going to introduce more changes elsewhere and I didn't want to confuse you. If it is good as I'm proposing, then I can make this change once you ok it. @junrao